### PR TITLE
MINIFICPP-1442 Uplift sol2 to latest release to support Lua version 5.4

### PR DESCRIPTION
**Pull request by lordgamez**
https://issues.apache.org/jira/browse/MINIFICPP-1442

---------------------------------------------------------
Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [ ] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically main)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI results for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: cmake/Sol2.cmake [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# Using file(DOWNLOAD) instead of FetchContent due to this issue in cmake versions older than 3.18 https://gitlab.kitware.com/cmake/cmake/-/issues/20526
+
+set(SOL2_INCLUDE_DIR "${CMAKE_BINARY_DIR}/_deps/sol2/" CACHE STRING "" FORCE)
+if(NOT EXISTS "${SOL2_INCLUDE_DIR}/sol.hpp")
```
Comment (adamdebreceni) [inline cmake/Sol2.cmake:21]: it seems to me that other libraries use `FetchContent`

Comment (lordgamez) [reply cmake/Sol2.cmake:21]: The problem with `FetchContent` is that if we want to use it for single files that are not archives we have to use the `
DOWNLOAD_NO_EXTRACT` option. Unfortunately due to an issue it does not copy it to the expected directory and leaves it in an unreachable subdirectory. The issue is fixed in CMake 3.18 but we only require 3.16. More information about this issue can be found here: https://gitlab.kitware.com/cmake/cmake/-/issues/20526

Comment (adamdebreceni) [reply cmake/Sol2.cmake:21]: oh now I see there is a comment about it

**Review by arpadboda — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -119,7 +118,7 @@ jobs:
           cmake -DUSE_SHARED_LIBS=ON -DCMAKE_BUILD_TYPE=Release -DSTRICT_GSL_CHECKS=AUDIT -DFAIL_ON_WARNINGS=ON -DENABLE_AWS=ON -DENABLE_AZURE=ON -DENABLE_BUSTACHE=ON -DENABLE_COAP=ON \
               -DENABLE_ENCRYPT_CONFIG=ON -DENABLE_GPS=ON -DENABLE_JNI=ON -DENABLE_LIBRDKAFKA=ON -DENABLE_LINTER=ON -DENABLE_MQTT=ON -DENABLE_NANOFI=ON -DENABLE_OPC=ON -DENABLE_OPENCV=ON \
               -DENABLE_OPENWSMAN=ON -DENABLE_OPS=ON -DENABLE_PCAP=ON -DENABLE_PYTHON=ON -DENABLE_SENSORS=ON -DENABLE_SFTP=ON -DENABLE_SQL=ON -DENABLE_SYSTEMD=ON -DENABLE_TENSORFLOW=OFF \
-              -DENABLE_USB_CAMERA=ON ..
+              -DENABLE_USB_CAMERA=ON -DENABLE_LUA_SCRIPTING=1 ..
```
Comment (szaszm) [inline .github/workflows/ci.yml:121]: I would change this to =ON (from =1) for consistency.

Comment (lordgamez) [reply .github/workflows/ci.yml:121]: Update in 0d1505d6245b56fbc17c7023aaf80aaf3631c52c

Comment (szaszm) [reply .github/workflows/ci.yml:121]: That's another instance in the file.

Comment (lordgamez) [reply .github/workflows/ci.yml]: Thanks, fixed it.

**Review by szaszm — APPROVED** [review]

**Review by adamdebreceni — APPROVED** [review]

File: cmake/Sol2.cmake [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    file(DOWNLOAD "https://github.com/ThePhD/sol2/releases/download/v3.2.2/sol.hpp" "${SOL2_INCLUDE_DIR}/sol/sol.hpp"
```
[the remark this answers is not in the document]
Comment (lordgamez) [reply cmake/Sol2.cmake:22]: Yes, in case it does not exist `file(DOWNLOAD` recursively creates the directory path.

**Review by adam-markovics — APPROVED** [review]

**Pull request closed without merging**